operators/src/source/gdal_source/loading_info.rs

```diff
@@ -54,6 +53,134 @@ impl MetaData<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>
         Box::new(self.clone())
     }
 }
+#[derive(PartialEq, Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct GdalMetadataFixedTimes {
+    time_steps: Vec<TimeInterval>,
+    time_steps_no_gaps: Vec<TimeInterval>,
+    params: GdalDatasetParameters,
+    result_descriptor: RasterResultDescriptor,
+    time_placeholders: HashMap<String, GdalSourceTimePlaceholder>,
+}
+
+impl GdalMetadataFixedTimes {
+    #[allow(unused)]
+    pub fn new(
+        time_steps: Vec<TimeInterval>,
+        params: GdalDatasetParameters,
+        result_descriptor: RasterResultDescriptor,
+        time_placeholders: HashMap<String, GdalSourceTimePlaceholder>,
+    ) -> Result<Self> {
+        let mut sorted_time_steps = time_steps;
+        sorted_time_steps.sort_by_key(TimeInterval::start);
+
+        let mut time_steps_no_gaps = sorted_time_steps.clone();
+
+        for (index, time) in sorted_time_steps.iter().enumerate() {
+            if index < sorted_time_steps.len() - 1
+                && time.end() < sorted_time_steps[index + 1].start()
+            {
+                time_steps_no_gaps.push(TimeInterval::new(
+                    time.end(),
+                    sorted_time_steps[index + 1].start(),
+                )?);
+            }
+        }
+
+        time_steps_no_gaps.sort_by_key(TimeInterval::start);
+
+        Ok(GdalMetadataFixedTimes {
+            time_steps: sorted_time_steps,
+            time_steps_no_gaps,
+            params,
+            result_descriptor,
+            time_placeholders,
+        })
+    }
+
+    fn create_temporal_slice(
+        &self,
+        step_interval: TimeInterval,
+        empty: bool,
+    ) -> Result<GdalLoadingInfoTemporalSlice> {
+        self.params
+            .replace_time_placeholders(&self.time_placeholders, step_interval)
```

Comment:
you only have to replace the timeholders if empty is false, so please check for that first and replace afterwards